lib/config/options.rb

```diff
@@ -31,47 +31,6 @@ def prepend_source!(source)
       @config_sources.unshift(source)
     end
 
-    def reload_env!
-      return self if ENV.nil? || ENV.empty?
-
-      hash = Hash.new
-
-      ENV.each do |variable, value|
-        separator = Config.env_separator
-        prefix = (Config.env_prefix || Config.const_name).to_s.split(separator)
-
-        keys = variable.to_s.split(separator)
-
-        next if keys.shift(prefix.size) != prefix
-
-        keys.map! { |key|
-          case Config.env_converter
-            when :downcase then
-              key.downcase.to_sym
-            when nil then
-              key.to_sym
-            else
-              raise "Invalid ENV variables name converter: #{Config.env_converter}"
-          end
-        }
-
-        leaf = keys[0...-1].inject(hash) { |h, key|
-          h[key] ||= {}
-        }
-
-        unless leaf.is_a?(Hash)
-          conflicting_key = (prefix + keys[0...-1]).join(separator)
-          raise "Environment variable #{variable} conflicts with variable #{conflicting_key}"
-        end
-
-        leaf[keys.last] = Config.env_parse_values ? __value(value) : value
-      end
-
-      merge!(hash)
-    end
-
-    alias :load_env! :reload_env!
```

Comment:
@pkuczynski `Config::Options#load_env!` and `Config::Options#reload_env!` are not documented at all, but are technically `public` methods.

As such, depending on whether or not we consider all visible methods to be part of the public API of this gem, this could be considered a breaking change and warrants a major version bump.

The change for users that use `#load_env!` or `#reload_env!` is simple: they should just replace those usages with `#reload!`, which is documented.